lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -68,17 +221,91 @@
   return canonicalizeLocaleList(runtime, locales);
 }
 
+vm::CallResult<std::u16string> localeListToLocaleString(
+    vm::Runtime *runtime,
+    const std::vector<std::u16string> &locales) {
+  // 3. Let requestedLocales be ? CanonicalizeLocaleList(locales).
+  vm::CallResult<std::vector<std::u16string>> requestedLocales =
+      canonicalizeLocaleList(runtime, locales);
+  if (LLVM_UNLIKELY(requestedLocales == vm::ExecutionStatus::EXCEPTION)) {
+      return vm::ExecutionStatus::EXCEPTION;
+  }
+
+  // 4. If requestedLocales is not an empty List, then
+  // a. Let requestedLocale be requestedLocales[0].
+  // 5. Else,
+  // a. Let requestedLocale be DefaultLocale().
+  std::u16string requestedLocale = requestedLocales->empty()
+      ? getDefaultLocale()
+      : std::move(requestedLocales->front());
+  // 6. Let noExtensionsLocale be the String value that is requestedLocale with
+  // any Unicode locale extension sequences (6.2.1) removed.
+  std::u16string noExtensionsLocale =
+      toNoUnicodeExtensionsLocale(requestedLocale);
+
+  // 7. Let availableLocales be a List with language tags that includes the
+  // languages for which the Unicode Character Database contains language
+  // sensitive case mappings. Implementations may add additional language tags
+  // if they support case mapping for additional locales.
+  // 8. Let locale be BestAvailableLocale(availableLocales, noExtensionsLocale).
+  // Convert to C++ array for bestAvailableLocale function
+  const std::vector<std::u16string> availableLocalesVector =
```

Comment:
```suggestion
  const std::vector<std::u16string>& availableLocalesVector =
```
`getAvailableLocalesVector()` returns the local static vector by const-ref, so why are we creating a new copy here?

Reply:
Good catch! Also marking [this](https://github.com/facebook/hermes/pull/616#discussion_r740584433) as unresolved.